lib/core/versions/latest/RequestHandler.ts

```diff
@@ -181,7 +181,7 @@ export default class RequestHandler implements IRequestHandler {
       const document = DocumentComposer.transformToExternalDocument(didState, did, published);
 
       // Status is different if DID is deactivated.
-      const didDeactivated = didState.nextRecoveryCommitmentHash === undefined;
+      const didDeactivated = document.didDocumentMetadata.deactivated;
```

Comment:
Note: the change on this line is not really necessary, but I thought it might be useful for readability to make use of the newly added `deactivated` property. This should be equivalent as before, since `transformToExternalDocument` (called above) returns `createDeactivatedResolutionResult(...)` when `didState.nextRecoveryCommitmentHash === undefined`:
https://github.com/decentralized-identity/sidetree/blob/c3171462b7ee945ac6209cb3a21a607332953299/lib/core/versions/latest/DocumentComposer.ts#L25-L30
... and `createDeactivatedResolutionResult` would set `deactivated: true` in the `didDocumentMetadata` object:
https://github.com/decentralized-identity/sidetree/pull/1178/files#diff-a5730fc426536637a041ba8b67308520ed5fdce4c785ebcfcfa8496c5c22ef4cR132

Reply:
I agree it would be more readable, but I think the intent here is not to depend on knowledge of the returned external document and rely on `didState as much as possible, to minimize chance of breakage of this code in case for whatever reason the data model of the external document is changed. Which is why document type is `any`, as the code does not make any assumption about its schema.

Reply:
Makes sense. Amended commit to remove this change: 8744f65f36b206b090c8e41c793c4d8ae3a8aa05